mlir-tensorrt/executor/lib/CAPI/Common/Common.cpp

```diff
@@ -129,6 +129,30 @@ MTRT_Status mtrtExecutableGetStorageView(MTRT_Executable executable,
   return mtrtStatusGetOk();
 }
 
+MTRT_Status mtrtExecutableGetNumDataSegments(MTRT_Executable executable,
+                                             int64_t *result) {
+  const Executable *exe = unwrap(executable);
+  *result = exe->getNumDataSegments();
+  return mtrtStatusGetOk();
+}
+
+MTRT_Status mtrtExecutableGetDataSegmentInfo(MTRT_Executable executable,
+                                             int64_t index,
+                                             MTRT_StringView *data,
+                                             MTRT_StringView *name) {
```

Comment:
Please add `assert(data && name && "expected valid data and name output arguments");`